junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/EngineIdValidator.java

```diff
@@ -56,15 +56,15 @@ private static boolean validateReservedIds(TestEngine testEngine) {
 		if (!engineId.startsWith("junit-")) {
 			return true;
 		}
-		if (engineId.equals("junit-jupiter")) {
+		if ("junit-jupiter".equals(engineId)) {
```

Comment:
Although I understand the intention of this PR, this particular change cannot possibly avoid a `NullPointerException`, since the preceding if-statement has already invoked a method on `engineId`.

Instead of making changes like this one, I think we should rather focus on specifying that `org.junit.platform.engine.TestEngine.getId()` may never return `null`, and we can consider introducing a `Preconditions.notNull(testEngine.getId())` check if we are concerned about that.

Reply:
See:

- 28981647939d280c19719107ef0e1403e66de25f
- 941f80d6a8cccdb1f08b69eedf061f41faa147e9

Reply:
@pixeebot @ryandens Here's some feedback on the [java/switch-literal-first](https://docs.pixee.ai/codemods/java/pixee_java_switch-literal-first) codemod